### ISSUE TITLE
Decrypt from clipboard extremeley temperamental #Oreo

### ISSUE THREAD

**Issue by ucwhatic** [open-keychain/open-keychain#2207]
in order to get openkeychain on android 8 to decrypt a PGP encrypted message I have to first:
1.**paste it into a text file** (I use totalcmd-edit)
2.**Remove any tags** (left by most clients on line2 after the ----GPG line 
3.**Make sure its spaced out correctly** (fix any funky stuff that looks out of place)
4.**Save the file as a .txt**

then and only then I will be able to import the file into openkeychain and decrypt it. 

TL;DR I have had zero success using the decrypt from clipboard option. And limited success decrypting anything at all. Have tried with imported keys and keys created using default parameters in app with same result.

* Android Version: Android 8.0.0
* OpenKeychain Version: latest on F droid repo
* From Google Play or F-Droid?: Fdroid

**Comment by ucwhatic** [comment]
**UPDATE** I had a sneaking suspicion my TL;DR was painting with too wide of a brush so i just tried to decrypt some text that was encrypted **using the same openkeychain app** and it worked. 

Therefore, it seems that errors only arise when I am decrypting a clipboard that was **originally copied using an app that is not Openkeychain** (Firefox, Orfox, StyleNote, etc.)

**Comment by Valodim** [comment]
The problem is that the clipboard tends to mangle encrypted text in a way that isn't easily and unambiguously recoverable. We recently added some experimental heuristics to improve this behavior for key import from clipboard. This worked out fairly well from the feedback I got, so I will probably add a similar mechanism to decryption from clipboard.

**Comment by bam80** [comment]
Sorry I'm new here. Could young please point the format clipboard content should be to successful import keys from it? Tried to paste the keys file "as is" and failed. Importing from that file worked, however.

**Comment by jbeece** [comment]
@Valodim from my testing and experience the clipboard issue is strictly limited to new lines, so either regex newlines to OKP accepted standard or expand the standard or newlines accepted by OKP.

I am a novice programmer, if you were to point me to the  file/line location of the code where newlines are interpreted I can take a crack at it.
